Platon/backend/app/workspace_system/views.py

```diff
@@ -645,6 +668,209 @@ def delete(user_id, self):
         else:
             return make_response(jsonify({'error': 'Input Format Error'}), 400)
 
+@workspace_system_ns.route("/milestone")
+class MilestoneAPI(Resource):
+    
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error', 404: 'Not found'})
+    @api.response(200, 'Success', milestone_list_model)
+    @api.expect(get_milestone_parser)
+    @login_required
+    @workspace_exists(param_loc='args',workspace_id_key='workspace_id')
+    def get(user_id, self):
+        '''
+            Get Milestones.
+        '''
+        form = GetMilestoneForm(request.args)
+        if form.validate():
+            workspace_id = form.workspace_id.data
+
+            # check if the workspace is public or not.
+            try:
+                workspaceSearch = Workspace.query.filter(Workspace.id == workspace_id).first()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            if workspaceSearch.is_private:
+                # check if current user is an active contributor.
+                try:
+                    contributionSearch = Contribution.query.filter((Contribution.workspace_id == workspace_id)&(Contribution.user_id == user_id)).first()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+                if contributionSearch is None:
+                    return make_response(jsonify({'error': 'User is not a contributor'}), 401)
+
+                if not contributionSearch.is_active:
+                    return make_response(jsonify({'error': 'User is not an active contributor currently'}), 401)
+            
+            try:
+                milestoneSearch = Milestone.query.filter(Milestone.workspace_id == workspace_id).all()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            # If there is no milestone in corresponding workspace, I'll return an empty list with 200 code.
+            # Since it is not some kind of an error, I thought 200 code is more appropriate.
+            if len(milestoneSearch) == 0:
+                return make_response(jsonify({'result': []}), 200)
+
+            return_list = []
+            for milestone in milestoneSearch:
+
+                try:
+                    creator_user = User.query.filter(User.id == milestone.creator_id).first()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+                if creator_user is None:
```

Comment:
If the user who created a milestone deleted their account, I do not think we should raise an error here. Because, the workspace still exists, and the rest of the people will use those milestone in that workspace. So, that the creator of a milestone is no longer existent does not mean that we should not return the milestone information.

Reply:
You're right, I've changed it with the latest commit.